ansible_runner/utils/__init__.py

```diff
@@ -201,10 +202,15 @@ def dump_artifacts(kwargs):
 
         kwargs['envvars']['ANSIBLE_ROLES_PATH'] = roles_path
 
-    obj = kwargs.get('playbook')
-    if obj and isplaybook(obj):
-        path = os.path.join(private_data_dir, 'project')
-        kwargs['playbook'] = dump_artifact(json.dumps(obj), path, 'main.json')
+    playbook = kwargs.get('playbook')
+    if playbook:
+        # Ensure the play is a list of dictionaries
+        if isinstance(playbook, MutableMapping):
```

Comment:
Wouldn't `Mapping` work just as well here? Not sure why we need the extra import.